httpx/_types.py

```diff
@@ -24,7 +26,7 @@
     from ._config import Proxy, Timeout  # noqa: F401
     from ._models import URL, Cookies, Headers, QueryParams, Request  # noqa: F401
 
-StrOrBytes = Union[str, bytes]
+StrOrBytes = TypeVar("StrOrBytes", str, bytes)
```

Comment:
Was this change required somehow? We specifically don't want to use type vars/generics for `HeaderTypes` and others — we initially used that but realized a `Union` better fits what we accept (i.e. "`str` _or_ `bytes`, I don't care").

We initially used `typing.AnyStr` as our "str or bytes" annotation (which this is equivalent to), but changed it because

Reply:
I tried `MutableMapping[StrOrBytes, StrOrBytes]`, but I had the incompatibility error. that's why I decided to change the `StrOrBytes`.
What should we do? split  `Dict[StrOrBytes, StrOrBytes]` in `HeaderTypes` to `MutableMapping[str, str], MutableMapping[bytes, bytes]`?

Reply:
Yeah, it's odd. I actually traced that back to a mypy limitation in `Mapping`/`MutableMapping` (the key type is invariant). Opened #993 with a tight fix inspired by yours here. :-)

Reply:
Nice!